Files/Filesystem/StorageHistory/Helpers/StorageHistoryHelpers.cs

```diff
@@ -0,0 +1,71 @@
+﻿using Files.Helpers;
+using System;
+using System.Threading.Tasks;
+
+namespace Files.Filesystem.FilesystemHistory
+{
+    public class StorageHistoryHelpers : IDisposable
+    {
+        #region Private Members
+
+        private IStorageHistoryOperations _storageHistoryOperations;
+
+        #endregion
+
+        #region Constructor
+
+        public StorageHistoryHelpers(IStorageHistoryOperations storageHistoryOperations)
+        {
+            this._storageHistoryOperations = storageHistoryOperations;
+        }
+
+        #endregion
+
+        #region Undo, Redo
+
+        public async Task<Status> Undo()
+        {
+            if (CanUndo())
+            {
+                App.StorageHistoryIndex--;
+                int index = EnumerableHelpers.FitBounds(App.StorageHistoryIndex, App.StorageHistory.Count);
+                return await this._storageHistoryOperations.Undo(App.StorageHistory[index]);
+            }
+            return Status.InProgress;
+        }
+
+        public async Task<Status> Redo()
```

Comment:
I am not sure we want to implement a redo feature.

Reply:
Well the default Windows 10 File Explorer has the undo feature.
Why discard it if it's already implemented and working. I'm certain a lot of users (including me) will find the Redo useful